tests/test_init_command.rs

```diff
@@ -100,18 +110,24 @@ fn test_init_git_external() {
     assert!(jj_path.join("working_copy").is_dir());
     assert!(repo_path.is_dir());
     assert!(store_path.is_dir());
-    let git_target_file_contents = std::fs::read_to_string(store_path.join("git_target")).unwrap();
-    assert!(git_target_file_contents
-        .replace('\\', "/")
-        .ends_with("/git-repo/.git"));
+    let unix_git_target_file_contents = std::fs::read_to_string(store_path.join("git_target"))
+        .unwrap()
+        .replace('\\', "/");
+    if bare {
+        assert!(unix_git_target_file_contents.ends_with("/git-repo"));
+    } else {
+        assert!(unix_git_target_file_contents.ends_with("/git-repo/.git"));
+    }
 
     // Check that the Git repo's HEAD got checked out
     let stdout = test_env.jj_cmd_success(&repo_path, &["log", "-r", "@-"]);
-    insta::assert_snapshot!(stdout, @r###"
-    ◉  mwrttmoslwzp git.user@example.com 1970-01-01 01:02:03.000 +01:00 my-branch HEAD@git 8d698d4a8ee1
-    │  My commit message
-    ~
-    "###);
+    insta::allow_duplicates! {
+        insta::assert_snapshot!(stdout, @r###"
+        ◉  mwrttmoslwzp git.user@example.com 1970-01-01 01:02:03.000 +01:00 my-branch HEAD@git 8d698d4a8ee1
+        │  My commit message
+        ~
+        "###);
+    }
```

Comment:
> Note: added allow_duplicates to the init test so make it work, but it seems the original code didn't really assert anything using insta (since the snapshot files are not checked in) - not sure what the intention there is.

`insta` also allows defining the snapshot in the source code. That's what we use here. The second argument to `assert_snapshot!()` is the snapshot. `cargo insta review` and `cargo insta test --review` are able to update the snapshot in the code.

Reply:
ah, I must have been confused by some .snap files that were generated at some point, probably when the test was failing for other reasons